### PR TITLE
drivers/w5100/w5100.c: fix TX_WR register's value

### LINKED ISSUE #16417: w5100 driver's improvement
Last time, I found a bug in w5100  #15447, that will cause wrong pointer position when its tx ring buffer is full and cause wrong data to be sent.

I found there is anoter problem in drivers/w5100/w5100.c, in **send** function:
```
static int send(netdev_t *netdev, const iolist_t *iolist)
{
    w5100_t *dev = (w5100_t *)netdev;
    int sum = 0;
    spi_acquire(dev->dev->p.spi, dev->p.cs, SPI_CONF, dev->p.clk);
    uint16_t pos = raddr(dev, S0_TX_WR0, S0_TX_WR1);

    if (pos == 0) {
        pos = S0_TX_BASE;
    }

    for (const iolist_t *iol = iolist; iol; iol = iol->iol_next) {
        size_t len = iol->iol_len;
        pos = tx_upload(dev, pos, iol->iol_base, len);
        sum += len;
    }

    waddr(dev, S0_TX_WR0, S0_TX_WR1, pos);
     ......
```
should be replaced by
```
static int send(netdev_t *netdev, const iolist_t *iolist)
{
    w5100_t *dev = (w5100_t *)netdev;
    int sum = 0;
    spi_acquire(dev->dev->p.spi, dev->p.cs, SPI_CONF, dev->p.clk);
    uint16_t tx_wr = raddr(dev, S0_TX_WR0, S0_TX_WR1);
    uint16_t pos = (tx_wr & S0_MASK) + S0_TX_BASE;

    if (pos == 0) {
        pos = S0_TX_BASE;
    }

    for (const iolist_t *iol = iolist; iol; iol = iol->iol_next) {
        size_t len = iol->iol_len;
        pos = tx_upload(dev, pos, iol->iol_base, len);
        sum += len;
    }

    waddr(dev, S0_TX_WR0, S0_TX_WR1, tx_wr + sum);
     ......
```
Values in TX_WR register should be in range of 0 ~ 0xFFFF, if I don't modify this send function,
it will cause the value of TX_WR to vary between 0 ~ 0x2000.
This bug will cause about 5ms delay every time the send ring buffer is full.

### PR DESCRIPTION
**Pull request by liyue75**
### Contribution description

drivers/w5100/w5100.c
fix w5100's TX_WR register's value. Let the value exceed 0x2000.

### Issues/PRs references

Fixes #16417

### CODE REVIEW

**Comment by fjmolinas** [comment]
I've looked at the datasheet and the fix is sensible, but I do not have the BOARD to test @liyue75 do you think you could post some test output to show that everything is still working and ideally that the mentioned issue is fixed?

**Comment by MrKevinWeiss** [comment]
Ping @liyue75 

@fjmolinas if there is no answer is it better to have this in or not?

**Review by benpicco — APPROVED** [review]

**Comment by maribu** [comment]
Sorry, somehow it was forgotten to hit the merge button. Since some time has passed, I re-triggered the CI just to be sure not conflicts have emerged in the meantime.

@liyue75: Thanks for the fix. Feel welcome to remind us of hitting the merge button / providing a review etc. Sometimes we just loose track of all the PRs / issues we a working on.

**Comment by maribu** [comment]
@liyue75: Do you want to create a backport PR, once this PR is merged? You can do so by running

```
$ ./dist/tools/backport_pr/backport_pr.py --comment 16474
```

in the root of the repository. (The 16474 is the number of this PR, so for future backports just replace that.) You'll find the dependencies (that is, the python3 modules needed by the script) in `dist/tools/backport_pr/requirements.txt`. You will also have to place a GitHub API token in `~/.riotgithubtoken`. There is a `README.md` in that folder giving more details.

I can also run the script - but the backport PR will then be attributed to me for technical reasons. (Still, the commit will retain original authorship and future release will be branched from the main development branch, which references this PR and correctly attributes authorship.)

**Pull request merged**
